joblib/test/test_pickle.py

```diff
@@ -0,0 +1,21 @@
+"""
+Test that pickle and dill output the same bytes
+"""
+import pickle
+
+import nose
+
+from joblib.test.common import dill, with_dill, with_py3
+
+
+def global_func():
+    pass
+
+
+@with_py3
+@with_dill
+def test_dumps_eq():
+    # Test that dill and pickle produce the same results
```

Comment:
One option would be to have the `joblib.hash` function could still be based on classical pickle and maybe the `joblib.dump` / `joblib.load` functions too and only use dill for the `delayed` function of `joblib.Parallel` calls.